Add support for detecting dev-mode in Craft CMS v4

Caroga opened the pull request:
### Description

I've spent way too long on trying to debug why Vite wasn't working when I found out that it was actually the plugin that wasn't properly configured. Changing the env var solved it. 

**Changes**

- Added support for detecting dev-mode in Craft CMS v4 in default config.php and the documentation: `App::env('ENVIRONMENT') === 'dev' || App::env('CRAFT_ENVIRONMENT') === 'dev'`

Please let me know if you agree or not. Thanks a bunch for this plugin !

### Related issues
none

Caroga committed:
Update config.php instruction craftcms 4.x

Update config.php instruction to mention the correct environment variable for Craft CMS 4.x

khalwat commented:
So... environment variables can be named whatever you want. In this case, yes, that is what the default Craft 4 boilerplate uses... but plenty of people are still using Craft 3 which set it to just `ENVIRONMENT` so probably it should use both.

Something like:

```php
'useDevServer' => App::env('ENVIRONMENT') === 'dev' || App::env('CRAFT_ENVIRONMENT') === 'dev',
```

You up for changing the PR to that so I can merge it in?

Caroga committed:
Added support for detecting dev-mode in Craft CMS v4

Caroga commented:
I've implemented your suggestion, which in hindsight would have been better to start with. In my defense I was up late 😅 But happy to help out, please review my changes and let me know if I missed anything.
Thanks!

P.s. Updated the title and my first comment to reflect the changes better.

khalwat commented:
Looks great, thanks for the PR!
